cmd/kubefwd/services/services.go

```diff
@@ -244,12 +248,36 @@ Try:
 		}
 
 		// check connectivity
-		err = checkConnection(clientSet, namespaces)
+		err = checkConnectivity(clientSet)
 		if err != nil {
 			log.Fatalf("Error connecting to k8s cluster: %s\n", err.Error())
 		}
 		log.Infof("Succesfully connected context: %v", ctx)
 
+		// If a wildcard namespace selector was given, query for all the namespaces and use all non-kube-internal ones
+		if len(namespaces) == 1 && namespaces[0] == "*" {
```

Comment:
Is it too special to just make a special case for *?

I think the best way is to filter the namespace through the Field selectors and LabelSelector of the namespace.

If it is not flexible enough, a regular match(regular expression) for namespaces is also a good choice. (In this case, you need to list all namespaces and do regular matching. In some cases, too many namespaces may cause some impact on the cluster.) So if you use regular matching(regular expression), you should need a special flag to indicate regular opening.

It's just my little suggestion.

Reply:
In our case we don't have any labels or fields on our namespaces, the developers have access to everything running in the cluster, and for most of them they are also not involved in the organisation of the namespaces within k8s. They just need to be able to run applications which will need to talk to whatever is in that application's settings (that included the namespace in the service addresses). So for them having to have the responsibility of having to type out the namespaces is not ideal as they don't know them, we want them to be able to use whatever is present.

A regex flag seems equally good as an alternative implementation though and does give more flexibility. I'll have a look at going that way.

Reply:
I've switched it to a regex argument.